email_alert_service/models/email_alert.rb

```diff
@@ -20,13 +20,32 @@ def format_for_email_api
       "subject" => document["title"],
       "body" => EmailAlertTemplate.new(document).message_body,
       "tags" => strip_empty_arrays(document.fetch("details", {}).fetch("tags", {})),
-      "links" => strip_empty_arrays(document.fetch("links", {})),
+      "links" => strip_empty_arrays(document.fetch("links", {}).merge('taxons_tree' => taxon_tree)),
       "document_type" => document["document_type"]
     }
   end
 
 private
 
+  def taxon_tree
```

Comment:
I've got a method for this in https://github.com/alphagov/govuk_taxonomy_helpers that you can reuse for this. 

I think it would reduce these two methods to:
```ruby
document.taxons.flat_map(&:tree)
```

You might need to stick a `.uniq` on the end just in case the thing is tagged to multiple overlapping parts of the taxonomy.

This might be a nice helper to add to the gem as well.

The only problem is all the existing code has required you too pass two separate API responses (get_content and get_expanded_links) whereas I assume the object you get from the message queue has all the data combined. But we could easily add another factory method to handle that. https://github.com/alphagov/govuk_taxonomy_helpers/blob/master/lib/govuk_taxonomy_helpers/publishing_api_response.rb